src/cpu/aarch64/matmul/acl_matmul_utils.cpp

```diff
@@ -24,59 +24,69 @@ namespace cpu {
 namespace aarch64 {
 
 using namespace alg_kind;
-using namespace cpu::matmul;
-using namespace format_tag;
 
 namespace acl_matmul_utils {
 
 status_t init_conf_matmul(acl_matmul_conf_t &amp, memory_desc_t &src_md,
-        memory_desc_t &wei_md, memory_desc_t &dst_md, memory_desc_t &bias_md,
-        const matmul_desc_t &md, const primitive_attr_t &attr) {
+        memory_desc_t &wei_md, memory_desc_t &dst_md, const matmul_desc_t &md,
+        const primitive_attr_t &attr) {
 
     const memory_desc_wrapper src_d(&src_md);
     const memory_desc_wrapper wei_d(&wei_md);
     const memory_desc_wrapper dst_d(&dst_md);
-    const memory_desc_wrapper bia_d(&bias_md);
 
-    matmul_helper_t helper(src_d, wei_d, dst_d);
+    cpu::matmul::matmul_helper_t helper(src_d, wei_d, dst_d);
     const dim_t M = helper.M();
     const dim_t N = helper.N();
     const dim_t K = helper.K();
-    const dim_t batch = helper.batch();
+    const dim_t dst_batch = helper.batch();
+    const dim_t src_batch = helper.src_batch();
+    const dim_t wei_batch = helper.wei_batch();
+
+    // ACL supports broadcast for 3D shapes, and 4D shapes
+    // for e.g when ab in abcd is 1x1
+    bool is_1x1 = ((src_batch / wei_batch == src_batch)
+            || (wei_batch / src_batch == wei_batch));
```

Comment:
The name is a bit misleading. Does this sound better:
```suggestion
    bool batch_ok = IMPLICATION(src_batch > 1, wei_batch == 1)
         && IMPLICATION(wei_batch > 1, src_batch == 1);
```